src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -1100,37 +1132,227 @@ private void PopulateParamsFromRestRequest(Dictionary<string, object?> parameter
             }
+            else if (parameters is List<IValueNode> listOfIValueNode)
+            {
+                // For the example createbooks mutation written above, the list value for `items` is interpreted as a List<IValueNode>
+                // i.e. items is a list of ObjectValueNode(s).
+                listOfIValueNode.ForEach(iValueNode => PopulateMutationFieldsToAuthorize(entityAndFieldsToAuthorize, schema, entityName, context, iValueNode, runtimeConfig));
+            }
+            else if (parameters is ObjectValueNode objectValueNode)
+            {
+                // For the example createbook mutation written above, the node for publisher field is interpreted as an ObjectValueNode.
+                // Similarly the individual node (elements in the list) for the reviews, authors ListValueNode(s) are also interpreted as ObjectValueNode(s).
+                ProcessObjectFieldNodesForAuthZ(context, entityName, schemaObject, objectValueNode.Fields, entityAndFieldsToAuthorize, runtimeConfig);
+            }
+            else if (parameters is ListValueNode listValueNode)
+            {
+                // For the example createbook mutation written above, the list values for reviews and authors fields are interpreted as ListValueNode.
+                // All the nodes in the ListValueNode are parsed one by one.
+                listValueNode.GetNodes().ToList().ForEach(objectValueNodeInListValueNode => PopulateMutationFieldsToAuthorize(entityAndFieldsToAuthorize, schema, entityName, context, objectValueNodeInListValueNode, runtimeConfig));
+            }
+        }
 
-            if (string.IsNullOrEmpty(role))
+        /// <summary>
+        /// Helper method to iterate over all the fields present in the input for the current field and add it to the dictionary
+        /// containing all entities and their corresponding fields.
+        /// </summary>
+        /// <param name="context">Middleware context.</param>
+        /// <param name="entityName">Name of the entity.</param>
+        /// <param name="schemaObject">Input object type for the field.</param>
+        /// <param name="fieldNodes">List of ObjectFieldNodes for the the input field.</param>
+        /// <param name="fieldsToAuthorize">Dictionary to store all the entities and their corresponding fields are stored.</param>
+        /// <param name="runtimeConfig">Runtime config.</param>
+        private void ProcessObjectFieldNodesForAuthZ(
+            IMiddlewareContext context,
+            string entityName,
+            InputObjectType schemaObject,
+            IReadOnlyList<ObjectFieldNode> fieldNodes,
+            Dictionary<string, HashSet<string>> fieldsToAuthorize,
+            RuntimeConfig runtimeConfig)
+        {
+            fieldsToAuthorize.TryAdd(entityName, new HashSet<string>());
+            string dataSourceName = GraphQLUtils.GetDataSourceNameFromGraphQLContext(context, runtimeConfig);
```

Comment:
will nested insertion work for multiple config files as well?

Reply:
I think so, yes. However, nested insertion would be possible for entities specified in the same config since in the `relationships` section of the config file, a relationship can only be defined with an entity present in the same config.

Reply:
We should add a test for that - attempt to do nested insert between entities across different configs leads to failure

Reply:
I think this validation, and the corresponding test should be added to `RuntimeConfigValidator` stage itself, where it prohibits the user to add relationship between entities across different config files. With that in place, the generated schema won't even contain such an invalid relationship. I had a chat with @rohkhann some weeks back, and he confirmed that this is a validation we need to have in place.

Reply:
Please create a separate issue for adding this cross config file validation + test.

Reply:
yes this makes sense, since we dont support cross database relationships.

Reply:
Created issue here: https://github.com/Azure/data-api-builder/issues/2078.